### DISCUSSION TITLE
using Rscript for track and intensity plotting

### DISCUSSION THREAD

**Discussion by AustinConaty** [dtcenter/METplus#1712] [Plot Generation] [answered]
Hi There,
I was revisiting TC tutorial and things were executing and producing expected results until I got to
this web page : https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-6-track-intensity/use-case-track-intensity-plotting where I couldn't seem to get R working as expected.  I loaded modules and copied the plot_tcmpr.R locally.
Since this is my first attempt at running any rscripts, I suspect I am missing something obvious.  If there is already a discussion on this and I missed seeing it, please point me there.  Thanks.  Running on a discover node at NCCS NASA GSFC. Thanks.

**Comment by bikegeek** [comment]
Hi Austin,

What error messages are you seeing when you are trying to run plot-tcmpr.R?  I am using one of the NCAR hosts which is using Rscript version 3.6.1

**Reply by AustinConaty** [reply]
I've reviewed the recorded tutorial and I'm going to try again.  I'm using R version 4.1.0.
Here's too much information probably: /gpfsm/dnb31/aconaty/METplus-4.0.0_Tutorial/output/track_and_intensity/tcmpr_plots'].out('/dev/null',append=False): non-zero exit status (returncode=1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/discover/swdev/sdrabenh/my_libs/metplus/METplus-4.0.0/metplus/util/met_util.py", line 159, in run_metplus
    new_commands = process.run_all_times()
  File "/discover/swdev/sdrabenh/my_libs/metplus/METplus-4.0.0/metplus/wrappers/tcmpr_plotter_wrapper.py", line 261, in run_all_times
    except produtil.run.ExitStatusException as ese:
NameError: name 'produtil' is not defined
08/01 08:33:58Z run-METplus-metplus.TCMPRPlotter: ERROR:  Fatal error occurred
08/01 08:33:58.917 metplus.TCMPRPlotter (met_util.py:187) INFO: Check the log file for more information: /gpfsm/dnb31/aconaty/METplus-4.0.0_Tutorial/output/track_and_intensity/logs/metplus.log.20220801083357
08/01 08:33:58.917 metplus (met_util.py:834) INFO: /gpfsm/dnb31/aconaty/METplus-4.0.0_Tutorial/output/track_and_intensity/metplus_final.conf: write metplus.conf here
08/01 08:33:58.929 metplus (met_util.py:218) ERROR: METplus has finished running but had 1 error.
08/01 08:33:58Z run-METplus-metplus: ERROR:  METplus has finished running but had 1 error.
08/01 08:33:58.929 metplus (met_util.py:219) INFO: Check the log file for more information: /gpfsm/dnb31/aconaty/METplus-4.0.0_Tutorial/output/track_and_intensity/logs/metplus.log.20220801083357

**Reply by bikegeek** [reply]
Is it possible to include the log file from your METplus run?  It looks like something isn't set up - I've seen the produtil error before but cannot remember what is the cause.  I hope that the log file will provide some more information.

**Comment by AustinConaty** [comment]
Hi,

At first, METplus wasn’t finding Rscript.  When I loaded local modules so
Rscript was added to the path, the script did attempt to execute, but it
seems I have an excessively large number of .tcst files in my request.

I think I’m getting closer.  I’ll try to get a more useful request submitted.

**Comment by bikegeek** [comment]
Hi Austin,

Do you have the log file for the run that produced the produtil error?

**Comment by bikegeek** [comment, marked as answer]
Austin,
one of the other engineers on the project suggested running METplus 4.1.0, which should provide output from the R script.  The produtil error you observed is due to a missing import of produtil.  This has been fixed in the next version of the METplus wrappers.  Can you please try again with the latest version of METplus?

**Comment by bikegeek** [comment]
I'll leave this discussion open for another week in case you have any new information.  Sometime next week I'll close this discussion and if this issue persists, you can open another discussion.
